### PR TITLE
KTIJ-891 False positive "replace `if` with `when`" quickfix when else-if expression does not check a single value

### PR DESCRIPTION
**Pull request by t-kameyama**
Issue link: <https://youtrack.jetbrains.com/issue/KTIJ-891>

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/intentions/branchedTransformations/branchedTransformationUtils.kt [hunk 1]
```diff
@@ -65,28 +59,32 @@ fun KtWhenExpression.getSubjectToIntroduce(checkConstants: Boolean = true): KtEx
...
+                }
+                else -> null
             }
-            else -> null
         }
-
+        else -> null
+    }
+    return getCandidate()?.takeIf {
+        it is KtNameReferenceExpression
```
Comment (yanex) [inline idea/src/org/jetbrains/kotlin/idea/intentions/branchedTransformations/branchedTransformationUtils.kt:84]: This doesn't make much sense. If all expressions excluding `KtThisExpression` match, the `it is KtNameReferenceExpression` check is useless. It looks like a mistake in De Morgan's law application.

**Comment by yanex** [comment]
It seems you replaced all tests with `i == <value>` checks. I think it makes sense to leave at least some of the existing tests as is, and introduce new ones if needed.

**Review by yanex — COMMENTED** [review]
It seems you replaced all tests with i == <value> checks. I think it makes sense to leave at least some of the existing tests as is, and introduce new ones if needed.

**Review by yanex — APPROVED** [review]

**Comment by t-kameyama** [comment]
bda9f4c4289acd89ba62c5c94b2b6fc414aaa9b8

**Pull request closed without merging**